src/util/uncaughtExceptionHandler.js

```diff
@@ -0,0 +1,105 @@
+  if (config.reportUncaughtException) {
+    logger.info('Reporting uncaught exceptions is enabled.');
+    process.once(uncaughtExceptionEventName, onUncaughtException);
+  } else {
+    logger.info('Reporting uncaught exceptions is disabled.');
+  }
+};
+
+
+exports.deactivate = function() {
+  process.removeListener(uncaughtExceptionEventName, onUncaughtException);
+};
+
+
+function onUncaughtException(uncaughtError) {
+  // because of the way Error.prepareStackTrace works and how error.stack is only created once and then cached it is
+  // important to create the JSON formatted stack trace first, before anything else accesses error.stack.
+  var jsonStackTrace = stackTraceUtil.getStackTraceAsJson(stackTraceLength, uncaughtError);
+  finishCurrentSpanAndReportEvent(uncaughtError, jsonStackTrace);
+  logAndRethrow(uncaughtError);
+}
+
+
+function finishCurrentSpanAndReportEvent(uncaughtError, jsonStackTrace) {
+  var spans = finishCurrentSpan(jsonStackTrace);
+  var eventPayload = createEventForUncaughtException(uncaughtError);
+  agentConnection.reportUncaughtExceptionToAgentSync(eventPayload, spans);
+}
+
+
+function createEventForUncaughtException(uncaughtError) {
+  var eventText = JSON.stringify(serializeError(uncaughtError));
+  return {
+    title: 'A Node.js process terminated abnormally due to an uncaught exception.',
+    text: eventText,
+    plugin: 'com.instana.forge.infrastructure.runtime.nodejs.NodeJsRuntimePlatform',
+    id: pidStore.pid,
+    timestamp: Date.now(),
+    duration: 1,
+    severity: 10
+  };
+}
+
+
+function finishCurrentSpan(jsonStackTrace) {
+  var currentSpan = cls.getCurrentSpan();
+  if (!currentSpan) {
+    return [];
+  }
+  currentSpan.error = true;
+  currentSpan.ec = 1;
+  currentSpan.d = Date.now() - currentSpan.ts;
+  currentSpan.stack = jsonStackTrace;
+  currentSpan.transmit();
+  return transmission.getAndResetSpans();
+}
+
+
+function logAndRethrow(err) {
+  // Remove all listeners now, so the final throw err won't trigger other registered listeners a second time.
+  var registeredListeners = process.listeners(uncaughtExceptionEventName);
+  if (registeredListeners) {
+    registeredListeners.forEach(function(listener) {
+      process.removeListener(uncaughtExceptionEventName, listener);
+    });
+  }
+  // eslint-disable-next-line max-len
+  logger.error('The Instana Node.js sensor caught an otherwise uncaught exception to generate a respective Instana event for you. Instana will now rethrow the error to terminate this process, otherwise the application would be left in an inconsistent state, see https://nodejs.org/api/process.html#process_warning_using_uncaughtexception_correctly. The next line on stderr will look as if Instana crashed your application, but actually the original error came from your application code, not from Instana. Since we rethrow the original error, you should see its stacktrace below (depening on how you operate your application and how logging is configured.)');
```

Comment:
❤️

Reply:
In java you can manipulate stacktraces to prevent such problems. maybe that possible in js as well?

Reply:
@CodingFabian we never have access to the stack trace I'm talking about there. We need to terminate the Node.js process and for this, we rethrow the error. The stack trace will then include the line from which we rethrow it, there's no way around that.

Reply:
yes thats what I meant. In java you can rethrow the original exception, so the line that rethrows would not appear
https://stackoverflow.com/questions/42754270/re-throwing-exception-in-nodejs-and-not-losing-stack-trace
seems like its not really possible in js